From f530158305b822829fe8139c8e699cddadd54956 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 2 Jan 2024 21:05:59 +0100
Subject: [PATCH] unix: Fix "Open in BambuStudio" button on MakerWorld
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Make the "Open in BambuStudio" button on MakerWorld work by informing
the system that BambuStudio is the application to handle bambustudio
URLs, and that it can handle URLs.

This fixes this error message visible in Firefox' console when clicking
the button on the website:
Prevented navigation to “bambustudio://open?file=https%3A%2F%2Fmakerworld.bblmw.com[...]bambu_scraper_grip_ams.3mf” due to an unknown protocol.
---
 src/platform/unix/BambuStudio.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/platform/unix/BambuStudio.desktop b/src/platform/unix/BambuStudio.desktop
index 8493077bf3da..b7e9b4d6ba01 100644
--- a/src/platform/unix/BambuStudio.desktop
+++ b/src/platform/unix/BambuStudio.desktop
@@ -2,10 +2,10 @@
 Name=BambuStudio
 GenericName=3D Printing Software
 Icon=BambuStudio
-Exec=bambu-studio %F
+Exec=bambu-studio %U
 Terminal=false
 Type=Application
-MimeType=model/stl;model/3mf;application/vnd.ms-3mfdocument;application/prs.wavefront-obj;application/x-amf;
+MimeType=model/stl;model/3mf;application/vnd.ms-3mfdocument;application/prs.wavefront-obj;application/x-amf;x-scheme-handler/bambustudio;
 Categories=Graphics;3DGraphics;Engineering;
 Keywords=3D;Printing;Slicer;slice;3D;printer;convert;gcode;stl;obj;amf;SLA
 StartupNotify=false
-- 
2.43.0

